File: LeetCode/cpp/946.cpp
class Solution {
public:
    bool validateStackSequences(vector<int> &pushed, vector<int> &popped) {
        stack<int> res;
        int i = 0, n = popped.size();
        for (auto &p:pushed) {
            res.push(p);
            while (!res.empty() && i < n && res.top() == popped[i]) {
                ++i;
                res.pop();
            }
        }
        return i == n;
    }
};

File: LeetCode/python/946.py
class Solution(object):
    def validateStackSequences(self, pushed, popped):
        """
        :type pushed: List[int]
        :type popped: List[int]
        :rtype: bool
        """
        res = []
        i, n = 0, len(popped)
        for p in pushed:
            res.append(p)
            while res and i < n and res[-1] == popped[i]:
                i += 1
                res.pop()
        return i == n